gateway/src/apicast/policy/maintenance_mode/maintenance_mode.lua

```diff
@@ -19,20 +22,54 @@ function _M.new(configuration)
   policy.message_content_type = default_message_content_type
 
   if configuration then
+    policy.maintenance_upstreams = configuration.upstreams
     policy.status_code = tonumber(configuration.status) or policy.status_code
     policy.message = configuration.message or policy.message
     policy.message_content_type = configuration.message_content_type or policy.message_content_type
   end
 
+  policy:load_condition(configuration)
   return policy
 end
 
-function _M:rewrite()
+function _M:load_condition(config)
+  if not config or not config.condition then
+    return
+  end
+
+  local operations = {}
+  for _, operation in ipairs(config.condition.operations or {}) do
+    table.insert( operations,
+      Operation.new(
+        operation.left,
+        operation.left_type or default_template_type,
+        operation.op,
+        operation.right,
+        operation.right_type or default_template_type))
+  end
+  self.condition = Condition.new( operations, config.condition.combine_op or default_combine_op)
+end
+
+function set_maintenance_mode(self)
   ngx.header['Content-Type'] = self.message_content_type
   ngx.status = self.status_code
   ngx.say(self.message)
 
   return ngx.exit(ngx.status)
 end
 
+function _M:access(context)
+  -- Before the condition was not set, so the maintenance mode when condition is
```

Comment:
In this past, this policy does not have condition. So by default, if condition is not set it set the maintenance mode.